pkgs/development/python-modules/colorize-pinyin/default.nix

```
@@ -0,0 +1,28 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  setuptools,
+}:
+buildPythonPackage rec {
+  pname = "colorize_pinyin";
+  version = "2.1.1";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-0qqa2uUOqaLVkEJxDugwtQIKF8Ba5cRVeGww4oS+j7k=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [ ];
+
+  meta = {
+    description = "search for chinese pinyin and wrap it with HTML.";
+    homepage = "https://pypi.org/project/colorize-pinyin/";
+    changelog = "https://pypi.org/project/colorize-pinyin/${version}";
+    license = lib.licenses.wtfpl;
+    maintainers = [ ];
```

you must add a maintainer: yourself (most preferably) and/or other willing folks (optionally)

---

also the [sole executable](https://github.com/ratijas/colorize_pinyin/blob/70694b1a8701323fa5fee2dd63df75c046ca58a7/setup.py#L47) must be set as `meta.mainProgram`
```suggestion
    maintainers = [ ];
    mainProgram = "colorize_pinyin";
```

---

done